src/services/blockchain/ethereum/ethereum-blockchain-service.ts

```diff
@@ -278,21 +278,24 @@ export class EthereumBlockchainService implements BlockchainService {
   }
 
   async _buildTransactionRequest(rootCid: CID): Promise<TransactionRequest> {
-    const rootStrHex = rootCid.toString(base16)
-    const hexEncoded = '0x' + (rootStrHex.length % 2 == 0 ? rootStrHex : '0' + rootStrHex)
-    logger.imp(`Hex encoded root CID ${hexEncoded}`)
-
     logger.debug('Preparing ethereum transaction')
     const baseNonce = await this.wallet.provider.getTransactionCount(this.wallet.address)
 
     if (!this.config.useSmartContractAnchors) {
+      const rootStrHex = rootCid.toString(base16)
+      const hexEncoded = '0x' + (rootStrHex.length % 2 == 0 ? rootStrHex : '0' + rootStrHex)
```

Comment:
```suggestion
      const hexEncoded = '0x' + uint8arrays.toString(rootCid.bytes.slice(4), 16)
```
I would do something like this to be sure